scene/gui/graph_node.cpp

```diff
@@ -459,7 +459,7 @@ void GraphNode::_shape() {
 }
 
 void GraphNode::set_slot(int p_idx, bool p_enable_left, int p_type_left, const Color &p_color_left, bool p_enable_right, int p_type_right, const Color &p_color_right, const Ref<Texture2D> &p_custom_left, const Ref<Texture2D> &p_custom_right) {
-	ERR_FAIL_COND(p_idx < 0);
+	ERR_FAIL_COND_MSG(p_idx < 0, vformat("Cannot set because Slot ID p_idx = %d is smaller than 0.", p_idx));
```

Comment:
Try ERR_FAIL_INDEX

Reply:
That was my first thought too. 

Let's look at following scenario where `ERR_FAIL_INDEX(m_index, m_size)` doesn't work.

1.When the godot editor starts with a scene containing graphnode, the editor uses this set_slot() method.
The `slot_info.size()` is basically 0 at this point. So,you can't use `slot_info.size()` as m_size param to fail_index macro.

The problem is: As of now,no function or variable in graphnode actually stores the actual number of slots. As of now,a slot is slot only if it has at least one enabled property(left/right).They are added to slot_info map,if they are only enabled.This is also the reason why `slot_info.size()` can't be used as m_size param to fail_index_macro in other methods too.

One way I see to use ERR_FAIL_INDEX() would be to create a function like `get_physical_slot_count()/get_actual_slot_count()  `which returns all the slots contained in a graphnode despite being enabled or not.

With our current approach,this would allow creating slots which aren't processed,but just stored in the memory.

So,it would be better if we create a function I suggested before,

Or,in update methods, remove elements from slot_info Map, if they are greater than number of actual possible slots.
I don't think this second approach is any better.because it needs to wait for update notification.This also allows creating slots with no error but are removed immediately.

Also, @pycbouh What do you think about this?

Reply:
Yes, you can't use `ERR_FAIL_INDEX` here because you have a map, not a set or a list or a vector. Indices that exist are arbitrary, so you can't compare the index passed here to the size.